tensorflow/python/ops/summary_ops_v2.py

```diff
@@ -197,14 +197,32 @@ class SummaryWriter(object):
   """Interface representing a stateful summary writer object."""
 
   @abc.abstractmethod
-  def set_as_default(self):
-    """Enables this summary writer for the current thread."""
+  def set_as_default(self, step=None):
+    """Enables this summary writer for the current thread.
+
+    For the `step` argument, see `tf.summary.experimental.set_step()`.
+
+    Args:
+      step: An `int64`-castable default step value, or `None`.
+        When not `None`, the current step is modified to the given value.
+        When `None`, the current step is not modified.
+    """
     raise NotImplementedError()
 
   @abc.abstractmethod
   @tf_contextlib.contextmanager
-  def as_default(self):
-    """Returns a context manager that enables summary writing."""
+  def as_default(self, step=None):
+    """Returns a context manager that enables summary writing.
+
+    For the `step` argument, see `tf.summary.experimental.set_step()`.
```

Comment:
Here and elsewhere: can we instead repeat the relevant sentences in these docstrings as well?

Also, can you add an example of how to use this argument?

Reply:
Done, I included all relevant information in the doc string, and added an example, which also shows that `step` is reverted when the context manager exits.

Reply:
@nfelt could you advise @foxik on how to add tests? Not having test coverage for public API is undesirable.

Reply:
The tests have been added.